cpp/src/plasma/io/basic_connection.cc

```diff
@@ -0,0 +1,276 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "plasma/io/basic_connection.h"
+#include "arrow/util/logging.h"
+
+#include <chrono>
+#include <memory>
+#include <thread>
+#include <utility>
+
+namespace plasma {
+namespace io {
+
+/// Connect a TCP socket.
+///
+/// @param socket The socket to connect.
+/// @param ip_address The IP address to connect to.
+/// @param port The port to connect to.
+/// @return Status.
+std::error_code TcpConnect(asio::ip::tcp::socket& socket,
```

Comment:
This is not currently used, right? Let's just remove it!

Reply:
I created this because we can use it under windows. Windows do not support unix domain sockets, so we can use tcp socket instead (or we can use UDP, since it would be more efficient).
And do you know how to build it with the CI under windows?